pkg/controller/cloneset/cloneset_controller.go

```diff
@@ -509,6 +507,11 @@ func (r *ReconcileCloneSet) truncateHistory(
 	return nil
 }
 
+// claimPods tries to take ownership of a list of Pods.
+//
+// It will reconcile the following:
+//   * Adopt orphans if the selector matches.
+//   * Release owned objects if the selector no longer matches.
```

Comment:
Is this new logic added by this change? I cannot find it here and there.

Reply:
The adopt/release logic are already implemented in function `manager.ClaimOwnedObjects`. What I did in this pr is changing the scope of the incoming pods (from pods owned by the cloneSet to all pods in the same namespace) which makes it possible to consume pods with no owner but matching the cloneSet selector.

Reply:
Sorry, I don't quite understand what "Release owned objects " mean here, do you mean the controller will remove the ownerreference from the Pod if the Pod's selector does not match with CS selector anymore?

Reply:
yes, you are right, this operation is called "release" in https://github.com/openkruise/kruise/blob/65a2e644190649731cd96fcc5f69332828f6115e/pkg/util/refmanager/ref_manager.go#L228

I think it is inspired by https://github.com/kubernetes/kubernetes/blob/979c644df2ad051015a8ccf6475e3476eda0cea0/pkg/controller/controller_ref_manager.go#L87

Reply:
btw, "Release owned objects" is introduced in #417 , this pr tries to support "Adopt orphans"